lua/tabnine/config.lua

```diff
@@ -7,7 +7,8 @@ function M.set_config(o)
 		accept_keymap = "<Tab>",
 		dismiss_keymap = "<C-]>",
 		debounce_ms = 800,
-		suggestion_color = { gui = "#808080", cterm = 244 },
+		suggestion_hl_group = "Comment",
+		suggestion_color = nil,
```

Comment:
We need a way to choose which will be picked up by default and which will override if it is specified.

If both have values by default then we'll need a 3rd config option to choose which one will be used.

Reply:
Perhaps we actually reverse the check in `autocmds.lua`, so if a highlight group is specified, that overrules suggestion_color, which would allow old configs to work the same, and we could leave the hl_group as a `nil` default, but if the user sets a hl_group, then we use that (ignoring suggestion_color), since no one should have the hl_group already in their config, that makes this an opt-in change, rather than opt-out. (I'm assuming a config like `tabnine.setup({})`, if it had values, then it would be unaffected)

Reply:
I'm more than open to feedback on this though, I'm just trying to keep backwards compatibility, and ensure that the update doesn't surprise anyone, unless they intentionally make changes to their config.

Reply:
If we keep the defaults as they are, this should be considered a breaking change, as it *could* greatly change the user experience (if they have the comment group highlighted differently than the default suggestion_color), however, with the changes suggested, it could avoid being a breaking change, as by default, nothing changes, and instead it must be opted-into

Reply:
@ChristianChiarulli @aarondill  I'm terribly sorry for the delay - it has been crazy month so far, didn't manage to look into this as much as I wanted.

@ChristianChiarulli is there any way to get colors from hl_group? this way users could just tweak their own config. something like (psaudo)`
suggestion_color={gui = api.nvim_get_hl_group("Comment").gui..`

On the personal perspective - I prefer different colors for comments and inline completions

Reply:
So there's a couple options here, my original attempt is to get the inline completion to look more native to the user's colorscheme if they don't have any config set. 

This is especially useful if the user is changing their colorscheme frequently and expects the inline text to follow. This is why I think it's a good default to choose something like comment.

You can get colors from a hl group like this:

```
  local normal_hl = vim.api.nvim_get_hl_by_name("Normal", true)
```

You could also highlight the text with your own hl group and link it to comment, then colorscheme devs could choose to include the hl group or users could override specifically with something like:

```
  vim.api.nvim_create_autocmd({ "Colorscheme" }, {
    callback = function()
      vim.api.nvim_set_hl(0, "TabnineCompletion", { fg = "#909090", bg = "#333333" })
    end,
  })
```

![image](https://github.com/codota/tabnine-nvim/assets/29136904/744c3ff9-26cd-45cf-bca9-cc240635eca7)

Maybe the codelens hl group would be better (or maybe the inlayhint one when that gets merged). Again I think using a predefined group which you could link to your own hl group is better because the default will follow the users colorscheme better.

And with this change if the user wants to set a static foreground color they can and it will override the default hl group. 

But I guess you could also just add the foreground text snippet to the readme and do it that way.

Reply:
Thanks @ChristianChiarulli your feedback means a lot!
Can you elaborate further on linking hl groupa? This concept is new to me

Reply:
Sure take a look at the following colorscheme:

https://github.com/ChristianChiarulli/onedark.nvim/blob/master/lua/onedark/theme.lua#L199

on the above line I'm linking the @lsp.variable hl group to the @variable treesitter hl group for consistency between syntax highlighting that treesitter provides and what the language server provides.

Some plugins create their own hl groups like bqf for instance then link them with builtin hl groups for a good default experience across colorschemes, that you or a colorscheme creator can override.

https://github.com/kevinhwang91/nvim-bqf 

From the README:

```
hi default link BqfPreviewFloat Normal
hi default link BqfPreviewBorder Normal
hi default link BqfPreviewCursor Cursor
hi default link BqfPreviewCursorLine CursorLine
hi default link BqfPreviewRange IncSearch
hi default link BqfPreviewBufLabel BqfPreviewRange
hi default BqfSign ctermfg=14 guifg=Cyan
```

Reply:
I fully support adding a hl group, but like I mentioned before, I personally just don't like the idea of changing the default from the defined colors, as it could suddenly change the appearance for any user that didn't explicitly define their own colors.